docker-compose.yml

```diff
@@ -0,0 +1,44 @@
+version: "3"
+services:
+  wca_on_rails:
+    build:
+      context: WcaOnRails
+    ports:
+      - 3000:3000
+    environment:
+      DATABASE_HOST: wca_db
+    # First, install Ruby and Node dependencies
+    # Next, wait for the database to be ready for queries via https://stackoverflow.com/a/53717981/2558618
+    # Load the database if it does not exist yet
+    # Start the server and bind to 0.0.0.0 (vs 127.0.0.1) so Docker's port mappings work correctly
+    command: >
+      bash -c 'bundle install && yarn install &&
+      while ! mysql --host=wca_db --user=root -e "SELECT 1" >/dev/null 2>&1; do
+        echo "Waiting for MySQL to be ready" && sleep 5 ;
+      done &&
+      echo "MySQL is ready!" &&
+      if [[ "`mysqlshow --user=root --host=wca_db wca_test`" =~ "wca_test" ]] ;
+        then echo "Populating development db, this will take a while" && bin/rake db:load:development ;
```

Comment:
I think long-term it would be great if the WCA could publish a docker volume that could be loaded into the MySQL container. That would mean it just takes as long as the database download instead of the very long clobbering process. I agree that it might be nice to optionally use the test db, but I'm not sure what the best way to do that would be. Something like an environment variable seems reasonable but that would have to be set every single time which is not ideal